src/ucp/core/ucp_types.h

```diff
@@ -61,6 +62,43 @@ typedef struct ucp_rkey_config_key      ucp_rkey_config_key_t;
 typedef struct ucp_proto                ucp_proto_t;
 
 
+/**
+ * UCP TL bitmap
+ *
+ * Bitmap type for representing which TL resources are in use.
+ */
+typedef ucs_bitmap_t(UCP_MAX_RESOURCES) ucp_tl_bitmap_t;
+
+
+/**
+ * Max possible value of TL bitmap (all bits are 1)
+ */
+extern const ucp_tl_bitmap_t ucp_tl_bitmap_max;
+
+
+/**
+ * Min possible value of TL bitmap (all bits are 0)
+ */
+extern const ucp_tl_bitmap_t ucp_tl_bitmap_min;
+
+
+#define UCT_TL_BITMAP_FMT          "0x%lx 0x%lx"
+#define UCT_TL_BITMAP_ARG(_bitmap) (_bitmap).bits[0], (_bitmap).bits[1]
```

Comment:
for consistency with other macros (e.g TL_RESOURCE_DESC_FMT) need to treat _bitmap as pointer to a bitmap struct